### PR TITLE
Add windows support

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,26 +10,33 @@ source:
...
     - exit $(test -f ${PREFIX}/lib/libfftw3f.a)          # [not win]
     - exit $(test -f ${PREFIX}/lib/libfftw3.a)           # [not win]
     - exit $(test -f ${PREFIX}/lib/libfftw3l.a)          # [not win]
     - exit $(test -f ${PREFIX}/lib/libfftw3f_threads.a)  # [not win]
     - exit $(test -f ${PREFIX}/lib/libfftw3_threads.a)   # [not win]
     - exit $(test -f ${PREFIX}/lib/libfftw3l_threads.a)  # [not win]
-    
-    # Verify headers are installed
+
+    # Verify headers are installde
```
Comment (jakirkham) [inline recipe/meta.yaml]: nit: Should be `installed` right?

**Comment by jakirkham** [comment]
How does this compare to PR ( https://github.com/conda-forge/fftw-feedstock/pull/25 )?

cc @grlee77

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,26 +10,33 @@ source:
   sha256: a5de35c5c824a78a058ca54278c706cdf3d4abba1c56b63531c2cb05f5d57da2
 
 build:
-  skip: True  # [win]
   number: 1
+  features:
+    - vc9  # [win and py27]
+    - vc14  # [win and py>=35]
+  skip: true  # [win and py36]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Slight preference for skipping `py35` instead as that will be dropped first.

**Comment by jschueller** [comment]
@jakirkham @grlee77 this is equivalent to #25, except threads are enabled, float/long double libs are also supported (not long  double on vc9 so I have  disabled it). Also there is a ctest part.

**Comment by jakirkham** [comment]
Neat! Thanks for the update. Sounds like a good idea to me.

Would note there has been discussion about an MSYS2-based package being added to `defaults`. Any thoughts on that in comparison to the VS build here?

**Comment by jschueller** [comment]
I guess it the difference is it would get the long double version working because it uses a good compiler unlike ms visual crap, but I wonder which package would use these. At least we'll have a version in conda-forge.

**Comment by grlee77** [comment]
This LGTM.  Two comments:

1.) Can you add ENABLE_SSE2 and ENABLE_AVX?  We have those options on for the existing packages (except for long double where they are not supported).

2.) The fact that long double libraries are missing is going to cause problems for pyFFTW as it currently requires all three sets of libraries.  There is a PR at https://github.com/pyFFTW/pyFFTW/pull/177 that should fix that, though.

**Comment by jschueller** [comment]
@grlee77 I added sse2/avx. For pyfftw, at worst we can patch if dynamic detection is not ready.

File: recipe/config.h.in [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+#define F77_FUNC_(name,NAME) name ## _
+
+/* Define if F77_FUNC and F77_FUNC_ are equivalent. */
+#define F77_FUNC_EQUIV 1
+
+/* Define if F77 and FC dummy `main' functions are identical. */
+/* #undef FC_DUMMY_MAIN_EQ_F77 */
+
+/* C compiler name and flags */
+#define FFTW_CC "gcc -std=gnu99 -O3 -fomit-frame-pointer -mtune=native -malign-double -fstrict-aliasing -fno-schedule-insns -ffast-math"
```
Comment (jakirkham) [inline recipe/config.h.in]: What's up with this line?

**Comment by jschueller** [comment]
This is hardwired, probably from a previous build

**Comment by jakirkham** [comment]
Yeah, I think this got copied from running `autoconf` on [`configure.ac`]( https://github.com/FFTW/fftw3/blob/fftw-3.3.6/configure.ac#L752 ). In any event, this has no meaning to us and we should drop it.

**Comment by jakirkham** [comment]
Upstream has been recently accepted PR ( https://github.com/FFTW/fftw3/pull/103 ), which adds an option to build with CMake. This allows for Windows support. Though they are sorting out some issues with the Windows build. Would be good if we can help them out. Expect that we will be using whatever Windows solution upstream settles on in the near future.

**Pull request merged**